shared/markdown/parser.pegjs

```diff
@@ -167,26 +166,26 @@ NativeEmoji
 LinkChar
  = !(SpecialChar+ (LineTerminatorSequence)) char:NonBlank { return char }
 
-Link 
+Link
   = url:( [([]* ("http"i "s"i? ":")? (LinkChar+) ) & {
-    let URL = [].concat.apply([], url).join('')
-    if (URL.length < 4) { // 4 chars is the shortest a URL can be (i.e. t.co)
+    // All words basically go into this function so lets not make it slow
+    const maybeQuote = url[0] || []
+    const maybeProtocol = url[1] || []
+    const maybeLink = [] = url[2] || []
+    if (maybeLink.length < 4) { // 4 chars is the shortest a URL can be (i.e. t.co)
       return null
     }
-
-    /* Make sure this is a real TLD */
-    const tldMatch = URL.match(tldExp)
-    if (tldMatch) {
-      const tld = tldMatch[4]
-      if (!tlds.includes(tld)) { // tlds is an array of all valid TLDs
+    const link = maybeLink.join('')
+    if (!tldExp.test(link)) { // includes a valid tld?
```

Comment:
old way did a weak regexp then did a search of the list, instead use a stronger regexp